### PR TITLE
Compatibility with typed constants in PHP 8.4

### LINKED ISSUE #1440: ReflectionClass::IS_READONLY is a typed class constant in PHP 8.4
PHP 8.4 made the ReflectionClass::IS_READONLY class constant typed in https://github.com/php/php-src/pull/12378. Since `Roave\BetterReflection\Reflection\Adapter\ReflectionClass` overrides it, the following exception is thrown when using PHP 8.4:

```
Type of Roave\BetterReflection\Reflection\Adapter\ReflectionClass::IS_READONLY must be compatible with ReflectionClass::IS_READONLY of type int
```

### PR DESCRIPTION
**Pull request by kukulich**
Fixes https://github.com/Roave/BetterReflection/issues/1440

### CODE REVIEW

**Comment by Ocramius** [comment]
Closes #1441

File: src/Reflection/Adapter/ReflectionClass.php [hunk 1]
```diff
@@ -33,7 +33,7 @@
  */
 final class ReflectionClass extends CoreReflectionClass
 {
-    public const IS_READONLY = 65536;
+    public const IS_READONLY_COMPATIBILITY = 65536;
```
Comment (Ocramius) [inline src/Reflection/Adapter/ReflectionClass.php:37]: Let's please mark this `@internal`

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: src/Reflection/Adapter/ReflectionClass.php (hunk 1 above)
Comment (kukulich) [reply src/Reflection/Adapter/ReflectionClass.php:37]: Done.

**Comment by Ocramius** [comment]
Thanks @kukulich!

**Pull request merged**